### PR TITLE
Calculate vouchers using base prices

### PR DESCRIPTION
**Pull request by stnatic**
I want to merge this change because...

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by korycins — APPROVED** [review]

**Comment by korycins** [comment]
@stnatic Can you confirm that the new flow takes into account a discount?

File: saleor/checkout/base_calculations.py [hunk 1]
```diff
@@ -135,26 +135,28 @@ def calculate_base_line_total_price(
...
 
 def calculate_base_price_for_shipping_method(
     checkout_info: "CheckoutInfo",
-    shipping_method_info: ShippingMethodInfo,
+    shipping_method_info: "ShippingMethodInfo",
     lines=None,
-) -> TaxedMoney:
+) -> Money:
     """Return checkout shipping price."""
+    from .fetch import CheckoutLineInfo
```
[the remark this answers is not in the document]
Comment (stnatic) [reply saleor/checkout/base_calculations.py:158]: Yes, I'm getting a circular import without it

File: saleor/discount/utils.py [hunk 2]
```diff
@@ -225,8 +223,9 @@ def validate_voucher_in_order(order: "Order"):
     customer_email = order.get_customer_email()
     if not order.voucher:
         return
+    value = subtotal.gross if display_gross_prices() else subtotal.net
```
[the remark this answers is not in the document]
Comment (stnatic) [reply saleor/discount/utils.py:226]: I moved it from here: https://github.com/saleor/saleor/pull/9237/files#diff-a9d95a35b8a29a0f6ff91190a38077cb4def3b7286d5997812773f61618cdb2cL138

TBH I'm puzzled and I'm not sure what's the difference between those two

**Comment by stnatic** [comment]
@korycins this PR does not change the flow yet, discount amount is included in the "base calculations" for checkout's total. some additional math might be needed later on if taxes/discounts are swapped places inside of `fetch_checkout_prices_if_expired`

File: saleor/discount/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/discount/utils.py:226]: I would change it to `include_taxes_in_prices`, I believe `display_gross_prices` is a flag that was useful in Saleor1.0

**Review by IKarbowiak — APPROVED** [review]

File: saleor/discount/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/discount/utils.py:226]: They are two separate settings:
- `display_gross_prices` was deciding whether frontend should render gross or net prices (from what I see it's still used in some places in logic), it was only about display from what I recall
- `include_taxes_in_prices` - this was more about internal calculations, whether they should use nets or gross

**Review by maarcingebala — APPROVED** [review]

**Review by L3str4nge — APPROVED** [review]

**Comment by korycins** [comment]
@stnatic Have you checked this changes with https://github.com/saleor/saleor-plugin-avatax-excise ?

**Review by tomaszszymanski129 — APPROVED** [review]

**Comment by stnatic** [comment]
I haven't tested end to end with the excise plugin yet, but it seems to override the same `PluginManager` methods as the original Avatax plugin. The renamed method for calculating subtotal was introduced recently in a shipping-related ticket, so it isn't really used anywhere across Saleor yet

**Pull request merged**